src/backend/distributed/utils/foreign_key_relationship.c

```diff
@@ -144,6 +145,30 @@ GetForeignKeyConnectedRelationIdList(Oid relationId)
 }
 
 
+/*
+ * ConnectedToReferenceTableViaFKey returns true if given relationId is
+ * connected to a reference table via its foreign key subgraph.
+ */
+bool
+ConnectedToReferenceTableViaFKey(Oid relationId)
+{
+	/* we can't rely on foreign key graph */
+	InvalidateForeignKeyGraph();
+
+	List *fkeyConnectedRelations = GetForeignKeyConnectedRelationIdList(relationId);
```

Comment:
We somewhere have a function which does this loop?

Reply:
we have a loop in `CascadeOperationForConnectedRelations` but it's slightly different than this one

Reply:
I meant we can use `RelationIdListHasReferenceTable(fkeyConnectedRelations)`

Reply:
true, I will address  that